NeoML/src/TraditionalML/BytePairEncoderTrainer.cpp

```diff
@@ -17,22 +17,34 @@ limitations under the License.
 #pragma hdrstop
 
 #include <NeoML/TraditionalML/BytePairEncoderTrainer.h>
+#include <NeoML/TraditionalML/SubwordEncoder.h>
 #include <BytePairEncoder.h>
 
 namespace NeoML {
 
-// Special tokens.
-static const CString StartOfWordTokenInternal( "/\xFF" );
-static const CString EndOfWordTokenInternal( "\\\xFF" );
+// Start-of-Word token for the internal dictionary
+static const CString SowToken( "//\xFF" );
+// End-of-Word token for the internal dictionary
+static const CString EowToken( "\\\\\xFF" );

```

Comment:
Two pairs of [S|E]ow constants are placed in two distinct files.
I suggest to add a compact class, which provides access to special tokens depending on params (UseRawBytes, etc)

Reply:
Also we want to continue using the deprecated pair of constants for utf-8 bpe mode.

Reply:
Sow/Eow are now being serialized along with the dictionary, a user can set up their own if an external vocabulary is used. For the pipeline "Trainer->Encoder", the Trainer now sets special tokens. Encoder doesn't have its own, the only purpose of the deprecated pair is to load older models.

Reply:
For your saved models, the old constants will be used. Is there any difference between new and old tokens for your future models? I still have doubts if there any point in customizable Eow/Sow in Trainer (we have bytemode now)

Reply:
Motivation for using the old constants: a significant number of tokens of completed bpe contain EoW as a subtoken, so we should try to avoid excessively long EoW tokens whenever possible.

Reply:
I have reverted EoW/BoW to their shorter versions since we expect strings (inputs) to be in utf-8 in any mode. If there are some complaints, custom EoW/BoW setters will be added to the trainer